### PR TITLE
EU-24-gc-upload

### PR DESCRIPTION
**Pull request by jenjou**

### CODE REVIEW

File: encode_utils/gc_storage.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Downloads file as bytes. We bypass the `read` interface so we don't mess with
+        `self.pos`.
+        """
+        return self.blob.download_as_string()
+
+    def read_json(self) -> Dict[str, Any]:
+        """
+        Read file and convert to JSON
+        """
+        return json.loads(self.read_bytes().decode())
```
Comment (paul-sud) [inline encode_utils/gc_storage.py:141]: Missing newline here

File: encode_utils/gc_storage.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    def md5sum(self) -> str:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def size(self) -> int:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_uri_without_scheme(self) -> str:
```
Comment (paul-sud) [inline encode_utils/gc_storage.py:34]: Don't think we need this method.

File: encode_utils/gc_storage.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        raise NotImplementedError
+
+    @abstractmethod
+    def read_bytes(self) -> bytes:
+        raise NotImplementedError
+
+    @abstractmethod
+    def read_json(self) -> Dict[str, Any]:
+        raise NotImplementedError
+
```
Comment (paul-sud) [inline encode_utils/gc_storage.py:51]: Don't need these methods either.

File: encode_utils/gc_storage.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    @property
+    def size(self) -> int:
+        return self.blob.size
+
+    @staticmethod
+    def b64_to_hex(value: str) -> str:
+        return b64decode(value).hex()
+
+    def get_uri_without_scheme(self) -> str:
+        return f"{self.blob.bucket.name}/{self.blob.name}"
```
Comment (paul-sud) [inline encode_utils/gc_storage.py:106]: Can delete this method.

File: encode_utils/gc_storage.py (hunk 1 above)
Comment (paul-sud) [inline encode_utils/gc_storage.py:141]: Can delete these methods.

**Review by paul-sud — CHANGES_REQUESTED** [review]
Should delete some unused methods, but looks good otherwise.

File: encode_utils/gc_storage.py (hunk 1 above)
Comment (jenjou) [reply encode_utils/gc_storage.py]: Removed several methods following comments.

**Pull request merged**